pkg/controller/redissnapshot/redissnapshot_controller.go

```diff
@@ -261,3 +269,53 @@ func (r *ReconcileRedisSnapshot) createSnapshot(ctx context.Context, cacheSvc el
 	r.logger.Info(msg)
 	return croType.PhaseInProgress, croType.StatusMessage(msg), nil
 }
+
+func buildRedisSnapshotInfoMetricLabels(cr *integreatlyv1alpha1.RedisSnapshot, clusterID, snapshotName string) map[string]string {
+	labels := buildRedisSnapshotGenericMetricLabels(cr, clusterID, snapshotName)
+	if len(string(cr.Status.Phase)) != 0 {
+		labels["statusPhase"] = string(cr.Status.Phase)
+		return labels
+	}
+	return labels
+}
+
+func buildRedisSnapshotGenericMetricLabels(cr *integreatlyv1alpha1.RedisSnapshot, clusterID, snapshotName string) map[string]string {
+	labels := map[string]string{}
+	labels["clusterID"] = clusterID
+	labels["resourceID"] = cr.Name
+	labels["namespace"] = cr.Namespace
+	labels["instanceID"] = snapshotName
+	labels["productName"] = cr.Labels["productName"]
+	labels["strategy"] = redisProviderName
+	labels["statusPhase"] = ""
+	return labels
+}
+
+func (r *ReconcileRedisSnapshot) exposeRedisSnapshotMetrics(ctx context.Context, cr *integreatlyv1alpha1.RedisSnapshot) {
+	// build instance name
+	snapshotName := cr.Status.SnapshotID
+
+	// get Cluster Id
+	logrus.Info("setting redis snapshot information metric")
+	clusterID, err := resources.GetClusterID(ctx, r.client)
+	if err != nil {
+		logrus.Errorf("failed to get cluster id while exposing information metric for %v", snapshotName)
+		return
+	}
+
+	// build metric labels
+	infoLabels := buildRedisSnapshotInfoMetricLabels(cr, clusterID, snapshotName)
+
+	// build generic labels
+	genericLabels := buildRedisSnapshotGenericMetricLabels(cr, clusterID, snapshotName)
+
+	// set status gauge
+	resources.SetMetricCurrentTime(resources.DefaultRedisSnapshotMetricName, infoLabels)
```

Comment:
I guess the same changed can be applied here as with the postgres controller. We can label this metric with a merged map of `infoLabels` and `genericLabels`, and remove lines `315-320` underneath

Reply:
same as postgressnapshot , metric should be 0 or 1, will remove the time stamp.